Modules/socketmodule.c

```diff
@@ -5574,6 +5574,23 @@ static PyType_Spec sock_spec = {
     .slots = sock_slots,
 };
 
+static PyObject *
+socket_getattr(PyObject *self, PyObject *name)
+{
+    const char *attrname = PyUnicode_AsUTF8(name);
+    PyObject *sock_type = PyType_FromSpec(&sock_spec);
```

Comment:
Don't reinitialize the type every time the attribute is accessed. You could possibly call `PyObject_GetAttr` on `self` (which is the module object), but I'm not sure if that will be recursive. If it is, use `PyObject_GenericGetAttr` or store the type on the module state.

Reply:
Do you mean PyObject_SetAttr()?

Reply:
No, with this change, `SocketType` will get reinitialized every time it's accessed in `__getattr__`, right?

Reply:
Oh wait. I misread the code. It's plain wrong: you must not call PyType_FromSpec(). Otherwise you get a new type at each call and it's just plain wrong...

You should be able to get `state->sock_type` from `get_module_state(self)`.